### DISCUSSION TITLE
AnalysisTemplate for GCP managed prometheus

### DISCUSSION THREAD

**Discussion by poonsalai** [argoproj/argo-rollouts#3307] [Q&A]
Hi Team

In documentation (per below link) we have the example for Amazon Managed Prometheus but somehow we are unable to location any example for GCP managed prometheus..

https://argo-rollouts.readthedocs.io/en/stable/analysis/prometheus/#utilizing-amazon-managed-prometheus

When we are running a query onto the GCP console using PromQL it works fine and gives the result per below but it is failing when we run it using the AnalysisTemplate and we get 401 while doing it.. because we are not able to get any way to pass the authentication using AnalysisTemplate for GCP.

Our AnalysisTemplate looks like below :

<img width="868" alt="image" src="https://github.com/argoproj/argo-rollouts/assets/105905102/7ca6fa9d-69ae-4a22-aa94-9d65c09907d4">

When we run it we are getting 401 per below

<img width="1709" alt="image" src="https://github.com/argoproj/argo-rollouts/assets/105905102/78e39227-3b09-4716-9388-e49d9031a4fb">

Could you please suggest any way how do we pass the authentication bit for GCP in AnalysisTemplate.. Below is the snippet for passing Auth in Amazon managed prometheus, need same way the snippet for GCP.

<img width="1715" alt="image" src="https://github.com/argoproj/argo-rollouts/assets/105905102/f5cf731e-7cb0-437c-8672-78e416d1b618">

**Also wanted to know while testing when we deploy another app and our blue-green strategy is in place.. and the promotion is paused because this AnalysisTemplate is going to run.. do we have any way we can put the traffic as well by templates? because otherwise t is going to be a manual task??** 

Waiting for a reply that can help us moving forward, thanks a lot :)

Thanks

**Comment by zachaller** [comment]
GCP for prometheus is currently not support, I don't have access to a GCP account to develop on, but I would be open for a contribution to add GCP as a supported auth method.

**Reply by NITHIN-JOHN-GEORGE** [reply]
@zachaller I was also looking for the same feature since we are using GKE. Any workarounds till the feature is developed?

**Reply by NITHIN-JOHN-GEORGE** [reply]
I also don't see an option for Google cloud monitoring in analysis whereas AWS cloud watch is supported.

**Comment by AndresPinerosZen** [comment]
I think you'll have to deploy the Google Managed Prometheus Frontend and use it as an authentication proxy.

**Reply by NITHIN-JOHN-GEORGE** [reply]
Hi @AndresPinerosZen i didn't get you on this .

**Reply by NITHIN-JOHN-GEORGE** [reply]
GCP prometheus will have a auth method but we can't pass it as a secret here

**Reply by jasonumiker** [reply]
He means this - https://cloud.google.com/stackdriver/docs/managed-prometheus/query-api-ui#ui-prometheus

**Comment by kostis-codefresh** [comment]
See https://github.com/argoproj/argo-rollouts/pull/5012
